### PR TITLE
MWPW-154054 + MWPW-154192: Blank space caused by delayed modal anchor

### PR DESCRIPTION
**Pull request by mirafedas**
When inserting an anchor link for the delayed modal we need to hide not only the link itself but the wrapping `<p>` element, otherwise it brings unwanted blank space.

Resolves: 
[MWPW-154054](https://jira.corp.adobe.com/browse/MWPW-154054)
[MWPW-154192](https://jira.corp.adobe.com/browse/MWPW-154192)

**Test URLs:**

Before: 
- https://main--cc--adobecom.hlx.page/products/aftereffects/free-trial-download?instant=2024-08-19T14:00:00.000Z
- https://main--cc--adobecom.hlx.page/creativecloud/buy/students?instant=2024-08-19T14:00:00.000Z
- https://main--milo--mirafedas.hlx.page/drafts/mirafedas/delayed-modals/delayed-modal

After: 
- https://main--cc--adobecom.hlx.page/products/aftereffects/free-trial-download?instant=2024-08-19T14:00:00.000Z&milolibs=MWPW-154054-blank-space--milo--mirafedas
- https://main--cc--adobecom.hlx.page/creativecloud/buy/students?instant=2024-08-19T14:00:00.000Z&milolibs=MWPW-154054-blank-space--milo--mirafedas
- https://mwpw-154054-blank-space--milo--mirafedas.hlx.page/drafts/mirafedas/delayed-modals/delayed-modal

### CODE REVIEW

File: libs/features/personalization/personalization.js [hunk 1]
```diff
@@ -127,6 +127,10 @@ const createFrag = (el, url, manifestId) => {
   const a = createTag('a', { href }, url);
   if (manifestId) a.dataset.manifestId = manifestId;
   let frag = createTag('p', undefined, a);
+  const isDelayedModalAnchor = /#.*delay=/.test(href);
```
Comment (3ch023) [inline libs/features/personalization/personalization.js:130]: could you add a test case for it in personalization.js please? that it adds 'hide-block' class

**Review by 3ch023 — CHANGES_REQUESTED** [review]
please add a test case in unit tests for this

File: libs/blocks/modal/modal.js [hunk 2]
```diff
@@ -228,7 +228,6 @@ export function delayedModal(el) {
   const { hash, delay } = getHashParams(el?.dataset.modalHash);
   if (!delay || !hash) return false;
   isDelayedModal = true;
-  el.classList.add('hide-block');
```
[the remark this answers is not in the document]
Comment (mirafedas) [reply libs/blocks/modal/modal.js:231]: That does not exist, the delayed modal may be added only in the promo manifest

**Comment by Axelcureno** [comment]
Please sync with stage to have NALA tests pass @mirafedas

File: libs/features/personalization/personalization.js [hunk 3]
```diff
@@ -127,6 +127,10 @@ const createFrag = (el, url, manifestId) => {
...
+  if (isDelayedModalAnchor) {
```
Comment (narcis-radu) [inline libs/features/personalization/personalization.js:131]:
```suggestion
  if (isDelayedModalAnchor) frag.classList.add('hide-block');
```

**Review by 3ch023 — APPROVED** [review]

**Review by chrischrischris — APPROVED** [review]

**Review by narcis-radu — APPROVED** [review]

**Review by Axelcureno — APPROVED** [review]

**Comment by mokimo** [comment]
Disregard the last message fixed by https://github.com/adobecom/milo/pull/2581

**Pull request merged**